src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -313,15 +395,41 @@ EnumEraNames
 Enumerates all the era names of the specified locale and calendar, invoking the callback function
 for each era name.
 */
-bool EnumEraNames(Locale& locale, CalendarId calendarId, EnumCalendarInfoCallback callback, const void* context)
+bool EnumEraNames(Locale& locale, CalendarId calendarId, CalendarDataType dataType, EnumCalendarInfoCallback callback, const void* context)
 {
 	UErrorCode err = U_ZERO_ERROR;
-	DateFormatSymbols dateFormatSymbols(locale, GetCalendarName(calendarId), err);
+	const char* calendarName = GetCalendarName(calendarId);
+	DateFormatSymbols dateFormatSymbols(locale, calendarName, err);
 	if (U_FAILURE(err))
 		return false;
 
 	int32_t eraNameCount;
-	const UnicodeString* eraNames = dateFormatSymbols.getEras(eraNameCount);
+	const UnicodeString* eraNames;
+
+	if (dataType == EraNames)
+	{
+		eraNames = dateFormatSymbols.getEras(eraNameCount);
+	}
+	else if (dataType == AbbrevEraNames)
+	{
+		if (strcmp(calendarName, "gregorian") == 0)
+		{
+			// NOTE: On Windows, the EraName is "A.D." and AbbrevEraName is "AD".
+			// But ICU/CLDR only supports "Anno Domini", "AD", and "A".
+			// So returning getEras (i.e. "AD") for both EraNames and AbbrevEraNames.
+			eraNames = dateFormatSymbols.getEras(eraNameCount);
```

Comment:
The issue is in trying to align with what is returned on Windows.
On Windows, EraName = "A.D." and AbbrevEraName = "AD".
ICU/CLDR gives us "Anno Domini", "AD", and "A".  Which should we return for EraName and which should we return for AbbrevEraName?
Should we say EraName is "Anno Domini" and AbbrevEraName is "AD"?  - With this option at least AbbrevEraName matches the Windows behavior.
Or EraName is "AD" and AbbrevEraName is "A"?  - With this option nothing matches the Windows values.